src/core/atomic.d

```diff
@@ -190,6 +254,14 @@ else version( AsmX86_32 )
         //
         // +=   -=  *=  /=  %=  ^^= &=
         // |=   ^=  <<= >>= >>>=    ~=
+        static if( op == "+=" && __traits(isIntegral, T) && is(T==V1) ) {
```

Comment:
Try `pragma(msg, T, " == ", V1, " = ", is(T == V1), " (op: ", op, ")");` just on the function entry below `body`.

The `is(T == V1)` constraint will miss half of the cases where this optimization is applicable, e.g. `atomicOp!"+="(u8, 1)`.

Note that you can also do this, which is defined in terms of `int` promoted add with subsequent coercion to `ubyte`.
```d
    ubyte u8;
    int val = 1;
    u8 += val;
```

I think we can implement this like the following.
```d
size_t tmp = mod; // convert all operands to size_t

asm
{
    mov RAX, tmp;
    mov RDX, val;
}

static if (T.sizeof == 1) asm { lock; xadd[RDX], AL; }
else static if (T.sizeof == 2) asm { lock; xadd[RDX], AX; }
else static if (T.sizeof == 4) asm { lock; xadd[RDX], EAX; }
else static if (T.sizeof == 8) asm { lock; xadd[RDX], RAX; }

asm { mov tmp, RAX; }

return cast(T)(tmp + mod);
```